cysync/src/store/hooks/useToken.ts

```diff
@@ -94,18 +96,108 @@ export const useToken: UseToken = () => {
+            const numB = new BigNumber(b.displayValue);
+            return numB.comparedTo(numA);
+          })
+        );
+        break;
+      case 1:
+        setTokenData(
+          [...tokens].sort((a, b) => {
+            const numA = new BigNumber(a.displayValue);
+            const numB = new BigNumber(b.displayValue);
+            return numA.comparedTo(numB);
+          })
+        );
+        break;
+      case 2:
+        setTokenData(
+          [...tokens].sort((a, b) => {
+            const tokenA = COINS[a.coin.toLowerCase()].name;
+            const tokenB = COINS[b.coin.toLowerCase()].name;
+            return tokenA.localeCompare(tokenB);
+          })
+        );
+        break;
+      case 3:
+        setTokenData(
+          [...tokens].sort((a, b) => {
+            const tokenA = COINS[a.coin.toLowerCase()].name;
+            const tokenB = COINS[b.coin.toLowerCase()].name;
+            return tokenB.localeCompare(tokenA);
+          })
+        );
+        break;
+      case 4:
+        setTokenData(
+          [...tokens].sort((a, b) => {
+            const numA = new BigNumber(a.displayBalance);
+            const numB = new BigNumber(b.displayBalance);
+            return numB.comparedTo(numA);
+          })
+        );
+        break;
+
+      case 5:
+        setTokenData(
+          [...tokens].sort((a, b) => {
+            const numA = new BigNumber(a.displayBalance);
+            const numB = new BigNumber(b.displayBalance);
+            return numA.comparedTo(numB);
+          })
+        );
+        break;
+
+      case 6:
+        setTokenData(
+          [...tokens].sort((a, b) => {
+            const numA = new BigNumber(a.displayPrice);
+            const numB = new BigNumber(b.displayPrice);
+            return numB.comparedTo(numA);
+          })
+        );
+        break;
+
+      case 7:
+        setTokenData(
+          [...tokens].sort((a, b) => {
+            const numA = new BigNumber(a.displayPrice);
+            const numB = new BigNumber(b.displayPrice);
+            return numA.comparedTo(numB);
+          })
+        );
+        break;
+      default:
+        break;
+    }
+  };
+
+  const sortTokensByIndex: UseTokenValues['sortTokensByIndex'] = index => {
+    if (tokenData.length === 0) return;
+    sortTokenData(tokenData, index);
+  };
+
   const getAllTokensFromWallet = async (walletId: string, ethCoin: string) => {
     const res = await erc20tokenDb.getByWalletId(walletId, ethCoin);
     const tokens: string[] = [];
     res.forEach(coin => {
       tokens.push(coin.coin);
     });
     setTokenList(tokens);
-    setTokenData(await getTokensWithPrices(res));
+    const unsortedTokens = await getTokensWithPrices(res);
+    sortTokenData(unsortedTokens, 0);
   };
 
   useEffect(() => {
-    getAllTokensFromWallet(currentWalletId, currentEthCoin);
+    // We handle only Ethereum Mainnet ERC20 tokens
+    if (currentWalletId && currentEthCoin === 'eth')
```

Comment:
This check is not required because we will support eth ropsten tokens in future. 
Also, even without this check, the logic is correct. The database will have no entries for the erc20 tokens of eth ropsten hence the array will be empty. Which is fine.

Reply:
Check it now please